jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -263,56 +269,92 @@ public void setStatsFactory(StatsFactory statsFactory) {
     this.statsFactory = statsFactory;
   }
 
+  public Configuration withReporter(ReporterConfiguration reporterConfig) {
+    this.reporterConfig = reporterConfig;
+    return this;
+  }
+
+  public Configuration withSampler(SamplerConfiguration samplerConfig) {
+    this.samplerConfig = samplerConfig;
+    return this;
+  }
+
+  public Configuration withCodec(CodecConfiguration codecConfig) {
+    this.codecConfig = codecConfig;
+    return this;
+  }
+
+  public Configuration withTracerTags(Map<String, String> tracerTags) {
+    this.tracerTags = tracerTags;
```

Comment:
You should use a new map, like `new HashMap(tracerTags)`, so that the caller won't change *your* copy of the map.